src/metabase/query_processor/preprocess.clj

```diff
@@ -52,8 +53,11 @@
    #'qp.perms/remove-permissions-key
    #'qp.constraints/maybe-add-default-userland-constraints
    #'validate/validate-query
-   #'fetch-source-query/resolve-card-id-source-tables
+   #'fetch-source-query/resolve-source-cards
    #'expand-macros/expand-macros
+   ;; ↑↑↑ ALL MIDDLEWARE ABOVE THIS POINT WILL SEE MLV2 PMBQL QUERIES ↑↑↑
+   ;; ↓↓↓ ALL MIDDLEWARE BELOW THIS POINT WILL SEE LEGACY MBQL QUERIES ↓↓↓
+   #'qp.convert-to-legacy/convert-to-legacy
```

Comment:
Check it out everyone, this is how we can convert the QP to MLv2 a few chunks at a time. Each successive PR will just move these three lines farther and farther down the middleware stacc until we don't have any more pre-processing middleware to convert. Then we'll do the same with compilation and post-processing.

I suppose we could actually do multiple conversions back and forth if we want to convert different pieces of middleware independently.